cartography/intel/cve/feed.py

```diff
@@ -68,49 +70,46 @@ def _map_cve_dict(cve_dict: Dict[Any, Any], data: Dict[Any, Any]) -> None:
     cve_dict["startIndex"] = data["startIndex"]
 
 
+def _configure_session(session: Session) -> None:
+    retry_policy = Retry(
+        total=8,
+        connect=1,
+        backoff_factor=1,
+        status_forcelist=[429, 500, 502, 503, 504],
+        allowed_methods=["GET"],
+    )
+    session.mount("https://", HTTPAdapter(max_retries=retry_policy))
+    logger.info(f"Configured session with retry policy: {retry_policy}")
+
+
 def _call_cves_api(url: str, api_key: str | None, params: Dict[str, Any]) -> Dict[Any, Any]:
     totalResults = 0
-    sleep_time = DEFAULT_SLEEP_TIME
-    retries = 0
     params["startIndex"] = 0
     params["resultsPerPage"] = RESULTS_PER_PAGE
     headers = {}
     headers["Content-Type"] = "application/json"
     if api_key:
+        sleep_between_requests = DEFAULT_SLEEP_TIME
         headers["apiKey"] = api_key
     else:
-        sleep_time = DELAYED_SLEEP_TIME  # Sleep for 6 seconds between each request to avoid rate limiting
+        sleep_between_requests = DELAYED_SLEEP_TIME
         logger.warning(
-            f"No NIST NVD API key provided. Increasing sleep time to {sleep_time}.",
+            f"No NIST NVD API key provided. Increasing sleep time to {sleep_between_requests}.",
         )
     results: Dict[Any, Any] = dict()
 
     with requests.Session() as session:
```

Comment:
Instead of creating the session here, maybe we could accept the session as an argument to _call_cves_api like dependency injection. This way _configure_session() can be called somewhere else.

I'm not familiar with this code but from the snippet you pasted in your description, it looks like _call_cves_api is pretty complicated and does a lot of things, so if we need to gut it and redo it, I'm supportive.

Reply:
the _call_cves_api is now fairly simple now after this change:
https://github.com/cartography-cncf/cartography/blob/fcf6cb11682db1b20eb10f00b536d9a5cd8800b0/cartography/intel/cve/feed.py#L85-L113

about further optimization of the session handling, i guess the most optimal place is to initiate the session in the `start_cve_ingestion` function in the `__init__.py`, yes?

Reply:
> the _call_cves_api is now fairly simple now after this change:

Agree, you improved it a lot.

> about further optimization of the session handling, i guess the most optimal place is to initiate the session in the start_cve_ingestion function in the __init__.py, yes?

Yeah I think so

Reply:
I moved the session to the higher level and passing it into feed.py calls.
There are some more small refactoring improvements to the code in the __init__.py.

I thought of having a singleton session for simplicity but noticed all the calls can further be parallelized using asyncio at some point, and singleton session would require a lot of changes again if we ever wanted to do that. With this version, it should be simpler to do so.

Reply:
well, i turned on auto-merge, it flagged that it's still waiting for your review
then i clicked on update branch, and it updated the branch then merged..

if you want any  more changes, let me know i can do in another PR then.